### PR TITLE
JBPM-5994 - Advanced queries with integration tests

### PR DESCRIPTION
**Pull request by domhanak**
Adds new module for testing advanced search client API with tests.
JMS is ignored.
Contains  changes from [main PRQ](https://github.com/kiegroup/droolsjbpm-integration/pull/930) in order to see if the tests are OK.

Also, contains some fixes with imports from wrong package, removes leftover dependencies on search-api

### CODE REVIEW

File: kie-server-parent/kie-server-tests/pom.xml [hunk 1]
```diff
@@ -129,6 +130,21 @@
         <artifactId>kie-tomcat-integration</artifactId>
         <version>${version.org.kie}</version>
       </dependency>
```
Comment (domhanak) [inline kie-server-parent/kie-server-tests/pom.xml]: ahh this need o go to bom,, apologies

File: kie-server-parent/kie-server-tests/pom.xml [hunk 2]
```diff
@@ -129,6 +130,21 @@
...
+      <dependency>
```
Comment (domhanak) [inline kie-server-parent/kie-server-tests/pom.xml]: these need to be moved to bom

File: kie-server-parent/kie-server-tests/pom.xml [hunk 3]
```diff
@@ -129,6 +130,21 @@
...
+        <groupId>org.kie.server</groupId>
+        <artifactId>kie-server-services-jbpm-search</artifactId>
+        <version>${version.org.kie}</version>
+      </dependency>
```
Comment (domhanak) [inline kie-server-parent/kie-server-tests/pom.xml]: need to add to be added to BOM

File: kie-server-parent/kie-server-tests/pom.xml [hunk 4]
```diff
@@ -129,6 +130,21 @@
...
+      <dependency>
+        <groupId>org.kie.server</groupId>
+        <artifactId>kie-server-rest-jbpm-search</artifactId>
+        <version>${version.org.kie}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.server</groupId>
+        <artifactId>kie-server-jbpm-search-api</artifactId>
+        <version>${version.org.kie}</version>
+      </dependency>
```
Comment (domhanak) [inline kie-server-parent/kie-server-tests/pom.xml]: leftover, will be removed

**Comment by domhanak** [comment]
No exception when using JSON format with incomaptible types.  2of test are failures related to this.

**Comment by domhanak** [comment]
jenkins retest this

**Comment by domhanak** [comment]
@DuncanDoyle hey, could you take a look at this? 2 of the fails are related - JSON format with bad types, throws no exception at all, however JAXB and XSTREAM ones do.
We should fix that prior to merging (if you push it into your PRQ I can update this one with your changes and we can merge it with tests to make sure it is OK). 
Also, note that in my latest commit there are some changes to fix imports and some leftover dependencies so you don't need to fix those.

**Comment by domhanak** [comment]
@DuncanDoyle could you also reference PRQ to bom?

**Comment by DuncanDoyle** [comment]
[quote]could you also reference PRQ to bom?[/quote]

OK, I'm not sure if I understand that question ....

I'll look at the test failures.

**Comment by domhanak** [comment]
@DuncanDoyle I meant pull request with depenedencies into knowledge BOM, I thought it is needed for some modules

**Comment by DuncanDoyle** [comment]
Yes, there are a number of TODOs for that in the code. I'll create a PR for that one as well.

**Comment by domhanak** [comment]
Closing. Will send a new one when Advanced Search is merged to master

**Pull request closed without merging**